### PR TITLE
Make RemoteConnectionInfo accessible from a transport client call

### PR DESCRIPTION
**Pull request by imotov**
We would like to have access to this information from a plugin.

### CODE REVIEW

**Review by nik9000 — APPROVED** [review]

**Review by jasontedor — CHANGES_REQUESTED** [review]
This needs stronger justification.

File: server/src/main/java/org/elasticsearch/transport/RemoteConnectionInfo.java [hunk 1]
```diff
@@ -36,13 +36,13 @@
...
-    final int numNodesConnected;
-    final String clusterAlias;
-    final boolean skipUnavailable;
+    public final List<TransportAddress> seedNodes;
+    public final List<TransportAddress> httpAddresses;
+    public final int connectionsPerCluster;
+    public final TimeValue initialConnectionTimeout;
+    public final int numNodesConnected;
+    public final String clusterAlias;
+    public final boolean skipUnavailable;
```
Comment (javanna) [inline server/src/main/java/org/elasticsearch/transport/RemoteConnectionInfo.java]: let's add getters for these?

**Review by javanna — APPROVED** [review]
I left one comment, LGTM otherwise, this is also needed if one day we want to add support for remote_info to the high-level REST client.

**Comment by s1monw** [comment]
we can do this I don't have objections to make it accessible but I think it should not be advertised to be the vanilla way of accessing valid cluster names. If this is accessed from a transport client that is one thing. but if there is a plugin that needs that to validate cluster names they should extend `RemoteClusterAware` instead.

**Comment by imotov** [comment]
@s1monw thanks! `RemoteClusterAware` is what we actually need. I will close the PR.

**Pull request closed without merging**